### PR TITLE
Azure CLI task - certificate endpoint support

### PR DESCRIPTION
**Pull request by vincentdass**

### CODE REVIEW

File: Tasks/AzureCLIV1/package-lock.json [hunk 1]
```diff
@@ -95,33 +51,23 @@
...
       "requires": {
-        "glob": "6.0.4",
         "minimatch": "3.0.4",
         "mockery": "1.7.0",
-        "node-uuid": "1.4.8",
         "q": "1.5.1",
         "semver": "5.5.0",
         "shelljs": "0.3.0",
-        "vso-node-api": "0.6.1"
+        "uuid": "3.2.1"
```
Comment (vincentdass) [inline Tasks/AzureCLIV1/package-lock.json]: OSS - IP scanned version

File: Tasks/AzureCLIV1/azureclitask.ts [hunk 2]
```diff
@@ -98,11 +98,24 @@ export class azureclitask {
 
     private static loginAzureRM(connectedService: string): void {
         var servicePrincipalId: string = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalid", false);
-        var servicePrincipalKey: string = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalkey", false);
+        let authType: string = tl.getEndpointAuthorizationParameter(connectedService, 'authenticationType', true);
```
Comment (sachinma) [inline Tasks/AzureCLIV1/azureclitask.ts]: are you backfilling the auth-type into all endpoints? Should we handle if this field is not present?

File: Tasks/AzureCLIV1/azureclitask.ts [hunk 3]
```diff
@@ -98,11 +98,24 @@ export class azureclitask {
...
+        let cliPassword: string = null;
+        if(authType == "spnCertificate") {
+            tl.debug('certificate based endpoint');
+            let certificateContent: string = tl.getEndpointAuthorizationParameter(connectedService, "servicePrincipalCertificate", false);
+            cliPassword = path.join(tl.getVariable('Agent.TempDirectory') || tl.getVariable('system.DefaultWorkingDirectory'), 'spnCert.pem');
```
Comment (sachinma) [inline Tasks/AzureCLIV1/azureclitask.ts]: We need to delete the file after login

File: Tasks/AzureCLIV1/azureclitask.ts (hunk 2 above)
Comment (vincentdass) [reply Tasks/AzureCLIV1/azureclitask.ts]: if the field is not present, we fall-back to service principal key as default.

Comment (sachinma) [reply Tasks/AzureCLIV1/azureclitask.ts]: Got it. Missed the optional: true

File: Tasks/AzureCLIV1/azureclitask.ts (hunk 3 above)
Comment (hiyadav) [reply Tasks/AzureCLIV1/azureclitask.ts]: +1

Comment (vincentdass) [reply Tasks/AzureCLIV1/azureclitask.ts]: done. thanks

Comment (vincentdass) [reply Tasks/AzureCLIV1/azureclitask.ts]: Agent with > 2.115.1 saves the file in Agent.tempDirectory which removes the file after the task completion.
For others, I have added the process of removing the file manually.

**Pull request merged**